dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -154,18 +183,117 @@ public ResourceSupport findRel(HttpServletRequest request, @PathVariable String
+				return ControllerUtils.toResponseEntity(HttpStatus.CREATED, null, new Resource(result));
+			}
+			else {
+				return ControllerUtils.toResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR, null, new Resource(result));
+			}				
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+		}
+		return ControllerUtils.toEmptyResponse(HttpStatus.INTERNAL_SERVER_ERROR);
+	}	
+	
+	@RequestMapping(method = RequestMethod.PATCH, value = "/{id:\\d+}")
+	public ResponseEntity<ResourceSupport> patch(HttpServletRequest request, @PathVariable String apiCategory,
+			@PathVariable String model, @PathVariable Integer id, @RequestBody(required = true) JsonNode jsonNode) throws HttpRequestMethodNotSupportedException {
+		return patchInternal(request, apiCategory, model, id, jsonNode);
+	}
+
+	@RequestMapping(method = RequestMethod.PATCH, value = "/{uuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}}")
+	public ResponseEntity<ResourceSupport> patch(HttpServletRequest request, @PathVariable String apiCategory,
+			@PathVariable String model, @PathVariable UUID id, @RequestBody(required = true) JsonNode jsonNode) throws HttpRequestMethodNotSupportedException {
+		return patchInternal(request, apiCategory, model, id, jsonNode);
+	}
+	
+	public <ID extends Serializable> ResponseEntity<ResourceSupport> patchInternal(HttpServletRequest request, String apiCategory,
+			String model, ID id, JsonNode jsonNode) throws HttpRequestMethodNotSupportedException {
+		
+		checkModelPluralForm(apiCategory, model);
+		DSpaceRestRepository<DirectlyAddressableRestModel, ID> repository = utils.getResourceRepository(apiCategory, model);
+		DirectlyAddressableRestModel modelObject = null;
+		try {
+			JsonPatchConverter patchConverter = new JsonPatchConverter(mapper);
+			Patch patch = patchConverter.convert(jsonNode);
+			modelObject = repository.patch(request, apiCategory, model, id, patch);
+		} 
+		catch (RepositoryMethodNotImplementedException | PatchUnprocessableEntityException | PatchBadRequestException e) {
+			log.error(e.getMessage(), e);
+			throw e;
+		}
+		catch (Exception e) {
+			log.error(e.getMessage(), e);
+			return ControllerUtils.toEmptyResponse(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		DSpaceResource result = repository.wrapResource(modelObject);
+		//TODO manage HTTPHeader
+		return ControllerUtils.toResponseEntity(HttpStatus.OK, null, result);		
+		
+	}
 	
 	private <ID extends Serializable> ResourceSupport findRelEntryInternal(HttpServletRequest request, String apiCategory, String model,
 			String id, String rel, String relid, Pageable page, PagedResourcesAssembler assembler, String projection) {
 		checkModelPluralForm(apiCategory, model);
-		DSpaceRestRepository<RestModel, ID> repository = utils.getResourceRepository(apiCategory, model);
-		Class<RestModel> domainClass = repository.getDomainClass();
+		DSpaceRestRepository<DirectlyAddressableRestModel, ID> repository = utils.getResourceRepository(apiCategory, model);
```

Comment:
As noted in #1889, I'm still not a fan of calling this class `DirectlyAddressableRestModel`, as the `DirectlyAddressable` portion has very little meaning to me. I'm not sure of a better name yet.